eZ/Publish/Core/MVC/Symfony/Routing/UrlWildcardRouter.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\MVC\Symfony\Routing;
+
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\URLWildcardService;
+use eZ\Publish\Core\MVC\Symfony\Routing\Generator\UrlAliasGenerator;
+use Psr\Log\LoggerInterface;
+use Symfony\Cmf\Component\Routing\ChainedRouterInterface;
+use Symfony\Cmf\Component\Routing\RouteObjectInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\Matcher\RequestMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\RouteCollection;
+use Symfony\Component\Routing\Route as SymfonyRoute;
+
+class UrlWildcardRouter implements ChainedRouterInterface, RequestMatcherInterface
+{
+    public const URL_ALIAS_ROUTE_NAME = 'ez_urlalias';
+
+    /**
+     * @var \eZ\Publish\API\Repository\URLWildcardService
+     */
+    private $wildcardService;
+
+    /**
+     * @var \eZ\Publish\Core\MVC\Symfony\Routing\Generator\UrlAliasGenerator
+     */
+    private $generator;
+
+    /**
+     * @var \Symfony\Component\Routing\RequestContext
+     */
+    private $requestContext;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param \eZ\Publish\API\Repository\URLWildcardService $wildcardService
+     * @param \eZ\Publish\Core\MVC\Symfony\Routing\Generator\UrlAliasGenerator $generator
+     * @param \Symfony\Component\Routing\RequestContext $requestContext
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(
+        URLWildcardService $wildcardService,
+        UrlAliasGenerator $generator,
+        RequestContext $requestContext,
+        LoggerInterface $logger
+    ) {
+        $this->wildcardService = $wildcardService;
+        $this->generator = $generator;
+        $this->requestContext = $requestContext;
+        $this->logger = $logger;
```

Comment:
Can we inject $logger via setter?

Reply:
> Can we inject $logger via setter?

Why? I consider setters a bad practice.

Reply:
@adamwojs yes, introduced it in https://github.com/ezsystems/ezpublish-kernel/pull/2609/commits/68791bac3821e41dac5af1ea26f23f2c4185f63c.